apps/dav/lib/Controller/CalendarImportController.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\DAV\Controller;
+
+use InvalidArgumentException;
+use OCA\DAV\AppInfo\Application;
+use OCA\DAV\CalDAV\CalendarImpl;
+use OCA\DAV\CalDAV\Import\ImportService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\ApiRoute;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\AppFramework\Http\Attribute\OpenAPI;
+use OCP\AppFramework\Http\Attribute\UserRateLimit;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\Calendar\CalendarImportOptions;
+use OCP\Calendar\IManager;
+use OCP\IGroupManager;
+use OCP\IRequest;
+use OCP\ITempManager;
+use OCP\IUserManager;
+use OCP\IUserSession;
+
+class CalendarImportController extends OCSController {
+
+	public function __construct(
+		IRequest $request,
+		private IUserSession $userSession,
+		private IUserManager $userManager,
+		private IGroupManager $groupManager,
+		private ITempManager $tempManager,
+		private IManager $calendarManager,
+		private ImportService $importService,
+	) {
+		parent::__construct(Application::APP_ID, $request);
+	}
+
+	/**
+	 * Import calendar data
+	 *
+	 * @param string $id calendar id
+	 * @param array{format?:string, validation?:int<0,2>, errors?:int<0,1>, supersede?:bool, showCreated?:bool, showUpdated?:bool, showSkipped?:bool, showErrors?:bool} $options configuration options
+	 * @param string $data calendar data
+	 * @param string|null $user system user id
+	 *
+	 * @return DataResponse<Http::STATUS_OK|Http::STATUS_BAD_REQUEST|Http::STATUS_UNAUTHORIZED|Http::STATUS_INTERNAL_SERVER_ERROR, array{error?: string, time?: float, created?: array{items: list<string>, total: int<0,max>}, updated?: array{items: list<string>, total: int<0,max>}, skipped?: array{items: list<string>, total: int<0, max>}, errors?: array{items: list<string>, total: int<0, max>}}, array{}>
```

Comment:
```suggestion
	 * @return DataResponse<Http::STATUS_OK|Http::STATUS_BAD_REQUEST|Http::STATUS_UNAUTHORIZED|Http::STATUS_INTERNAL_SERVER_ERROR, array{error?: string, time?: float, created?: array{items: list<string>, total: non-negative-int}, updated?: array{items: list<string>, total: non-negative-int}, skipped?: array{items: list<string>, total: non-negative-int}, errors?: array{items: list<string>, total: non-negative-int}}, array{}>
```
You should also make `array{items: list<string>, total: non-negative-int}` a separate psalm type, so you can reuse it.

Reply:
Done